tests/test_routes.py

```diff
@@ -1,47 +1,47 @@
-"""
-TestShopcart API Service Test Suite
-
-Test cases can be run with the following:
-  nosetests -v --with-spec --spec-color
-  coverage report -m
-"""
-import os
-import logging
-from unittest import TestCase
-from unittest.mock import MagicMock, patch
-from flask_api import status  # HTTP Status Codes
-from service.models import db
-from service.routes import app, init_db
-
-######################################################################
-#  T E S T   C A S E S
-######################################################################
-class TestYourResourceServer(TestCase):
-    """ REST API Server Tests """
-
-    @classmethod
-    def setUpClass(cls):
-        """ This runs once before the entire test suite """
-        pass
-
-    @classmethod
-    def tearDownClass(cls):
-        """ This runs once after the entire test suite """
-        pass
-
-    def setUp(self):
-        """ This runs before each test """
-        self.app = app.test_client()
-
-    def tearDown(self):
-        """ This runs after each test """
-        pass
-
-    ######################################################################
-    #  P L A C E   T E S T   C A S E S   H E R E
-    ######################################################################
-
-    def test_index(self):
-        """ Test index call """
-        resp = self.app.get("/")
-        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+"""
+TestShopcart API Service Test Suite
+
+Test cases can be run with the following:
+  nosetests -v --with-spec --spec-color
+  coverage report -m
+"""
+import os
+import logging
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+from flask_api import status  # HTTP Status Codes
+from service.models import db
+from service.routes import app, init_db
+
+######################################################################
+#  T E S T   C A S E S
+######################################################################
+class TestYourResourceServer(TestCase):
+    """ REST API Server Tests """
+
+    @classmethod
+    def setUpClass(cls):
+        """ This runs once before the entire test suite """
+        pass
+
+    @classmethod
+    def tearDownClass(cls):
+        """ This runs once after the entire test suite """
+        pass
+
+    def setUp(self):
+        """ This runs before each test """
+        self.app = app.test_client()
+
+    def tearDown(self):
+        """ This runs after each test """
+        pass
+
+    ######################################################################
+    #  P L A C E   T E S T   C A S E S   H E R E
+    ######################################################################
+
+    def test_index(self):
+        """ Test index call """
+        resp = self.app.get("/")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)

```

Comment:
you need to add the test route here for the create shopcart, add item to shopcart

Reply:
Thats what I was thinking was missing. the code I started off with was all about creating a cart.. nothing about adding items. I understand now that adding items will need to be done in the routes.py file. working on it

Reply:
See everything from line 95+